File: Find Smallest Digit/FindSmallestDigit.java
public class FindSmallestDigit {
    public static void main(String[] args) {
        java.util.Scanner sc = new java.util.Scanner(System.in);
        System.out.print("Enter a number: ");
        int n = sc.nextInt();
        int temp = Math.abs(n);
        int smallest = 9;
        while (temp > 0) {
            int digit = temp % 10;
            if (digit < smallest) {
                smallest = digit;
            }
            temp = temp / 10;
        }
        System.out.println("Smallest digit: " + smallest);
    }
}

File: Find Smallest Digit/find_smallest_digit.c
#include <stdio.h>
#include <stdlib.h>
int main() {
    int n;
    printf("Enter a number: ");
    scanf("%d", &n);
    int temp = abs(n);
    int smallest = 9;
    while (temp > 0) {
        int digit = temp % 10;
        if (digit < smallest) {
            smallest = digit;
        }
        temp = temp / 10;
    }
    printf("Smallest digit: %d\n", smallest);
    return 0;
}
